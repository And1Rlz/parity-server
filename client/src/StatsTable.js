// @flow

import _ from 'lodash'
import React, { Component } from 'react'
import Griddle from 'griddle-react'
import Stats from './Stats'
import MoneyCell from './MoneyCell'

const STATS = [
  'Goals',
  'Assists',
  '2nd Assist',
  'D-Blocks',
  'Catches',
  'Completions',
  'Throwaways',
  'ThrewDrop',
  'Drops',
  'Salary'
]

const columns = [
  'Name',
  'Team',
  ...STATS
]

let columnsMeta = _.map(columns, (col, idx) => {
  return {
    columnName: col,
    order: idx + 1
  }
})

columnsMeta[STATS.length + 1] = {
  columnName: 'Salary',
  order: STATS.length + 1,
  customComponent: MoneyCell
}

type Props = {
  week: number,
  stats: Stats
}

export default class StatsTable extends Component {
  props: Props

  state: {
    week: number,
    stats: Stats
  }

  constructor (props: Props) {
    super(props)

    this.state = {
      week: this.props.week,
      stats: this.props.stats
    }
  }

  render () {
    // filter players who only have a salary for this week.
    let statsArray = _.filter(this.state.stats.toArray(), (player) => {
      return _.keys(player).length > 4
    })

    return (
      <Griddle
        results={statsArray}
        resultsPerPage={statsArray.length}
        tableClassName='highlight responsive-table'
        columns={columns}
        columnMetadata={columnsMeta}
        showFilter={true}
        showPager={false}
        useGriddleStyles={false}
        filterPlaceholderText='Search players or team ...'
      />
    )
  }
}
