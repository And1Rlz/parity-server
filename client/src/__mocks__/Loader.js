import Stats from '../Stores/Stats'

const loader = jest.genMockFromModule('../Loader')

let fetchWeeks = function () {
  return [1, 2, 3]
}

let fetchStats = function (weekNum: number) {
  let data = {
    'Kevin Hughes': {
      'Goals': 2,
      'Assists': 1,
      '2nd Assist': 1,
      '3rd Assist': 3,
      '4th Assist': 1,
      '5th Assist': 1,
      'D-Blocks': 1,
      'Completions': 19,
      'Throwaways': 1,
      'ThrewDrop': 0,
      'Catches': 17,
      'Drops': 0,
      'Pick-Ups': 5,
      'Pulls': 0,
      'Calihan': 0,
      'OPointsFor': 5,
      'OPointsAgainst': 5,
      'DPointsFor': 5,
      'DPointsAgainst': 7,
      'Team': 'Like a Boss',
      'SalaryDelta': 77000,
      'Salary': 1071666
    }
  }

  return new Stats(data)
}

loader.fetchWeeks = fetchWeeks
loader.fetchStats = fetchStats

module.exports = loader
