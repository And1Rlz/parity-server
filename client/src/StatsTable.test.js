import renderer from 'react-test-renderer'
import React from 'react'
import Loader from './Loader'
import StatsTable from './StatsTable'

jest.mock('./Loader')

test('StatsTable renders correctly', () => {
  let week = 1
  let stats = Loader.fetchStats(1)
  const tree = renderer.create(<StatsTable week={week} stats={stats} />).toJSON()
  expect(tree).toMatchSnapshot()
})
