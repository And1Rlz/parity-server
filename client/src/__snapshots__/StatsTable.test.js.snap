exports[`test StatsTable renders correctly 1`] = `
<div
  className="griddle">
  <div
    className="top-section"
    style={null}>
    <div
      className="griddle-filter"
      style={null}>
      <div
        className="filter-container">
        <input
          className="form-control"
          name="filter"
          onChange={[Function bound handleChange]}
          placeholder="Search players or team ..."
          type="text" />
      </div>
    </div>
    <div
      className="griddle-settings-toggle"
      style={null}>
      
    </div>
  </div>
  
  <div
    className="griddle-container"
    style={null}>
    <div
      className="griddle-body">
      <div
        onScroll={[Function bound gridScroll]}
        style={null}>
        <table
          className="highlight responsive-table"
          style={null}>
          <thead>
            <tr
              className={null}
              style={Object {}}>
              <th
                className=""
                data-title="Name"
                onClick={[Function anonymous]}
                style={Object {}}>
                <span>
                  Name
                </span>
              </th>
              <th
                className=""
                data-title="Team"
                onClick={[Function anonymous]}
                style={Object {}}>
                <span>
                  Team
                </span>
              </th>
              <th
                className=""
                data-title="Goals"
                onClick={[Function anonymous]}
                style={Object {}}>
                <span>
                  Goals
                </span>
              </th>
              <th
                className=""
                data-title="Assists"
                onClick={[Function anonymous]}
                style={Object {}}>
                <span>
                  Assists
                </span>
              </th>
              <th
                className=""
                data-title="2nd Assist"
                onClick={[Function anonymous]}
                style={Object {}}>
                <span>
                  2nd Assist
                </span>
              </th>
              <th
                className=""
                data-title="D-Blocks"
                onClick={[Function anonymous]}
                style={Object {}}>
                <span>
                  D-Blocks
                </span>
              </th>
              <th
                className=""
                data-title="Catches"
                onClick={[Function anonymous]}
                style={Object {}}>
                <span>
                  Catches
                </span>
              </th>
              <th
                className=""
                data-title="Completions"
                onClick={[Function anonymous]}
                style={Object {}}>
                <span>
                  Completions
                </span>
              </th>
              <th
                className=""
                data-title="Throwaways"
                onClick={[Function anonymous]}
                style={Object {}}>
                <span>
                  Throwaways
                </span>
              </th>
              <th
                className=""
                data-title="ThrewDrop"
                onClick={[Function anonymous]}
                style={Object {}}>
                <span>
                  ThrewDrop
                </span>
              </th>
              <th
                className=""
                data-title="Drops"
                onClick={[Function anonymous]}
                style={Object {}}>
                <span>
                  Drops
                </span>
              </th>
              <th
                className=""
                data-title="Salary"
                onClick={[Function anonymous]}
                style={Object {}}>
                <span>
                  Salary
                </span>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              className="standard-row"
              onClick={null}>
              <td
                className={undefined}
                onClick={[Function bound handleClick]}
                style={null}>
                
                Kevin Hughes
              </td>
              <td
                className={undefined}
                onClick={[Function bound handleClick]}
                style={null}>
                
                Like a Boss
              </td>
              <td
                className={undefined}
                onClick={[Function bound handleClick]}
                style={null}>
                
                2
              </td>
              <td
                className={undefined}
                onClick={[Function bound handleClick]}
                style={null}>
                
                1
              </td>
              <td
                className={undefined}
                onClick={[Function bound handleClick]}
                style={null}>
                
                1
              </td>
              <td
                className={undefined}
                onClick={[Function bound handleClick]}
                style={null}>
                
                1
              </td>
              <td
                className={undefined}
                onClick={[Function bound handleClick]}
                style={null}>
                
                17
              </td>
              <td
                className={undefined}
                onClick={[Function bound handleClick]}
                style={null}>
                
                19
              </td>
              <td
                className={undefined}
                onClick={[Function bound handleClick]}
                style={null}>
                
                1
              </td>
              <td
                className={undefined}
                onClick={[Function bound handleClick]}
                style={null}>
                
                0
              </td>
              <td
                className={undefined}
                onClick={[Function bound handleClick]}
                style={null}>
                
                0
              </td>
              <td
                className={undefined}
                onClick={[Function bound handleClick]}
                style={null}>
                <span>
                  $1,071,666
                </span>
              </td>
            </tr>
          </tbody>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</div>
`;
